### PR TITLE
Test: add test suite to ensure a RaftStorage impl works correctly

### PR DESCRIPTION
**Pull request by drmingdrmer**
##### Test: add test suite to ensure a RaftStorage impl works correctly
`openraft::testing::Suite::test_all(|| MemStore::new())` will run a full
test on every RaftStorage APIs with `MemStore`.

TODO: `testing::Suite` needs to test state machine APIs and snapshot APIs.

---

### CODE REVIEW

File: openraft/src/testing/suite.rs [hunk 1]
```diff
@@ -0,0 +1,1511 @@
...
+
+        let l = store.try_get_log_entries(0..).await?.len();
+        let last = store.try_get_log_entries(0..).await?.last().unwrap().clone();
+
+        assert_eq!(l, 10, "expected 10 entries to exist in the log");
+        assert_eq!(last.log_id, (2, 10).into(), "unexpected log id");
+        Ok(())
+    }
+
+    // pub async fn apply_single(builder: &B) -> anyhow::Result<()> {
```
Comment (ariesdevil) [inline openraft/src/testing/suite.rs:794]: Are these comments useful?

Comment (drmingdrmer) [reply openraft/src/testing/suite.rs:794]: Yes. To provide state machine tests requires other APIs to be provided by `RaftStorage`.
These tests will be enabled when these apis are ready.

The hard part is that the state machine is quite business-related. It's hard to extract abstraction out of it for testing.

**Review by ariesdevil — APPROVED** [review]
Totally nice~

File: openraft/src/testing/suite.rs [hunk 2]
```diff
@@ -0,0 +1,1511 @@
...
+/// Create a blank log entry for test
+fn blank<D: AppData>(term: u64, index: u64) -> Entry<D> {
+    Entry {
+        log_id: LogId::new(term, index),
+        payload: EntryPayload::Blank,
+    }
+}
+
+/// Block until a future is finished.
+/// The future will be running in a clean tokio runtime, to prevent an unfinished task affecting the test.
```
Comment (SteveLauC) [inline openraft/src/testing/suite.rs:1505]: I wonder if this is needed, looks like the current log tests can be executed within a single runtime sequentially

Comment (SteveLauC) [reply openraft/src/testing/suite.rs:1505]: Or maybe we can run them concurrently, every test has its own store and state machine, so they should not affect each other, though I don't think this is needed.

Comment (drmingdrmer) [reply openraft/src/testing/suite.rs:1505]: No it does not looks necessary anymore 🤔

Comment (SteveLauC) [reply openraft/src/testing/suite.rs:1505]: Get it, then I will proceed to make the log tests runtime-agnostic by turning `test_all()` and `test_store()` asynchronous:)

**Pull request merged**